components/pattern/src/multi_named.rs

```diff
@@ -434,38 +434,31 @@ impl<'a> MultiNamedPlaceholderPatternIterator<'a> {
         match self.store.find(|x| (x as usize) <= 0x07) {
             Some(0) => {
                 // Placeholder
-                let Some(&[lead, trail]) = self.store.as_bytes().get(0..2) else {
+                let Some((&[lead, trail], remainder)) = self
+                    .store
+                    .split_at_checked(2)
+                    .map(|(a, b)| (a.as_bytes(), b))
+                else {
                     return Err(MultiNamedPlaceholderError::InvalidStore);
                 };
```

Comment:
```suggestion
                let (&[lead, trail], remainder) = self
                    .store
                    .split_at_checked(2)
                    .ok_or(MultiNamedPlaceholderError::InvalidStore)
                    .map(|(a, b)| (a.as_bytes(), b))?;
```

Reply:
My code also matches the `&[u8]` into `[u8; 2]`, which your code doesn't. I believe this is the most concise way to do this, and the let-else is consistent with the rest of the code in this file.